SearchHistory/models.py

```diff
@@ -1,22 +1,24 @@
 from django.db import models
 from django.utils import timezone
+from account.models.professional import Professional
+from account.models.client import Client
 
 
 class SearchHistory(models.Model):
     search_id = models.BigAutoField(primary_key=True)
-    professional_id = models.PositiveIntegerField(null=False, blank=False)
-    client_id = models.PositiveIntegerField(null=False, blank=False)
+    professional_id = models.ForeignKey(Professional, on_delete=models.CASCADE)
+    client_id = models.ForeignKey(Client, on_delete=models.CASCADE)
     date = models.DateTimeField(default=timezone.now)
 
     class Meta:
         db_table = 'SearchHistory'
 
+    def __str__(self):
+        return f"{self.client_id} search {self.professional_id} [{self.date}]"
+
     @staticmethod
-    def get_5_last_professionals_search_by_client(client_id: int):
+    def get_5_last_professionals_search_by_client(client_id, expected_result=5):
```

Comment:
Consider changing the name of the method because currently it can return a certain number of professionals and not necessarily 5..

Also, when you use the function you will need to know how many professionals the client was looking for to determine what value you send to the variable expected_result.

consider performing a test within the method which checks the length of the list received for the client.
And if its size is at least 5, then return the first five.
else, return the maximum possible.

Reply:
fixed